### DISCUSSION TITLE
how to run local to create pr

### DISCUSSION THREAD

**Discussion by reitenth** [sbondCo/Watcharr#430] [Q&A] [answered]
i would like to try my hand at helping development. how do i run local in vscode on windows? thanks

**Comment by IRHM** [comment, marked as answer]
Hey @reitenth, these steps should work:

1. Clone the code:
    ```
    git clone https://github.com/sbondCo/Watcharr.git && \
    cd Watcharr
    ```
2. Download frontend dependencies by running `npm i` in project root folder.
3. Download server dependencies by running `go get .` in the server folder.
4. In the projects root folder again, open up two terminals. In the first, you can run the server with `npm run server` and in the second, you can start a development frontend server with `npm run dev`.

Note: the frontend dev server (vite) will automatically update when you make changes, but the server needs a manual restart for changes to be reflected.

Edit: Created a guide for this on the docs site: https://watcharr.app/docs/installation/for-development

**Reply by reitenth** [reply]
thanks for the writeup! everything runs smoothly until i get to create an admin account. i cant create an account, it keeps saying network error.

**Reply by IRHM** [reply]
@reitenth That's weird, in devtools network tab, where are the api requests going?

They should go to `http://127.0.0.1:3080/api/<whichever endpoint>` (controlled by the `./src/lib/util/api.ts` file, baseURL variable).

If they are, are there any specific errors under the request or is it just timing out?

**Reply by reitenth** [reply]
There is one api request going to 127.0.0.1:3080/api/auth/available. here is the error.

`Cross-Origin Request Blocked: The Same Origin Policy disallows reading the remote resource at http://127.0.0.1:3080/api/auth/available. (Reason: CORS request did not succeed). Status code: (null).`

**Reply by IRHM** [reply]
@reitenth Hm not a big fan of CORS because it confuses me :(, looks like that specific error isn't directly related to CORS though, but instead a network error of some kind: https://developer.mozilla.org/en-US/docs/Web/HTTP/CORS/Errors/CORSDidNotSucceed

The page mentions:

> In many cases, it is caused by a browser plugin (e.g. an ad blocker or privacy protector) blocking the request. 

Could be worth trying in an incognito/private tab without any plugins, or in another browser if you have one installed.



Are you able to access your server at 127.0.0.1:3080 directly with a GET request (does loading that url in the browser directly give you a 404, or is it simply not accessible)?

**Reply by reitenth** [reply]
Loading the server directly gives me a "Unable to connect, can't establish a connection to the server in Firefox."

I tried an incognito tab in Firefox but I still get a network error on login with the same CORS error.

I tried in Microsoft Edge and also get an error.

`Uncaught (in promise) 
AxiosError
code
: 
"ERR_NETWORK"
config
: 
{transitional: {…}, adapter: Array(2), transformRequest: Array(1), transformResponse: Array(1), timeout: 0, …}
message
: 
"Network Error"
name
: 
"AxiosError"
request
: 
XMLHttpRequest {onreadystatechange: null, readyState: 4, timeout: 0, withCredentials: false, upload: XMLHttpRequestUpload, …}
stack
: 
"AxiosError: Network Error\n    at XMLHttpRequest.handleError (http://localhost:5173/node_modules/.vite/deps/axios.js?v=3be8b103:1450:14)\n    at Axios.request (http://localhost:5173/node_modules/.vite/deps/axios.js?v=3be8b103:1780:41)"
[[Prototype]]
: 
Error`

**Reply by IRHM** [reply]
@reitenth Looks like we can't connect to the server at `127.0.0.1:3080`. Do you think you could restart it (Ctrl+C in server terminal, then `npm run server` again) and see if it's outputting any errors?

Could also try accessing `localhost:3080` instead, i don't know if there's a difference in how they'll be loaded by the browser.

**Reply by reitenth** [reply]
Ok, I restarted the `npm run server` and this is the result.

`> watcharr@1.36.0 server`
`> cd ./server && MODE=DEV go run .` 
`Invalid parameter - =DEV`

**Reply by IRHM** [reply]
oh @reitenth that is all my fault :((( I forgot, I don't think Windows supports setting environment variables like that.

I'll see if I can look something up for you, not too sure how we could pass the MODE environment variable on Windows.

**Reply by IRHM** [reply]
@reitenth I tried this in powershell (in server directory), which seems to work: 

First: `go env -w CGO_ENABLED=1`

Then: `go get .` (might not be necessary again, but just incase)

Then: `cmd /V /C "set MODE=DEV&& go run ."`

But then I had to install `gcc`, which I downloaded from here (i just downloaded `64+32-bit MinGW-w64 edition`): https://jmeubank.github.io/tdm-gcc/articles/2021-05/10.3.0-release
 (https://stackoverflow.com/a/43588236)

If it did make you get gcc, you just need to restart powershell after it installs, then run the last command to start the server again.

**Reply by reitenth** [reply]
That seemed to have done the trick! Thanks buddy! I have some frontend UI changes that I think would look good. 

Would you know if I have to keep the terminal window open?

**Reply by IRHM** [reply]
No problem @reitenth, happy to help! Yep, you'd have to keep them open whilst developing.

If it's only small stuff you can certainly go ahead, but because I hate wasting anyone's time, you can feel free to open an issue discussing the changes you'd like to make. Though of course, if you'd like, you can just make the changes, and we can discuss afterwards when you open a pull request.

Thanks for helping out on Watcharr!
